### DISCUSSION TITLE
Table detection in images

### DISCUSSION THREAD

**Discussion by saanvib13** [chezou/tabula-py#384] [Q&A]
I have a couple of images and I want to detect tables in them. I am using the tabula library for the same. Since tabula works with pdfs, I am converting a set images to a pdf and then executing the tabula library on it. However, the library is unable to detect the tables in the converted pdf. What could be the reason behind it and are there any other alternatives to make this work? I also tried using the paddleocr library but it does not natively support batch processing. What can be done to process multiple images simultaneously on the GPU so as to detect tables in them?

**Comment by chezou** [comment]
Since tabula-java doesn't support image-based PDF, so does tabula-py.
https://tabula-py.readthedocs.io/en/latest/faq.html#i-got-an-empty-dataframe-how-can-i-resolve-it

**Reply by saanvib13** [reply]
Thank you for your response!! Can you suggest an alternate way or library that could support image based PDF?

**Reply by chezou** [reply]
Sorry, I don't know much about it.

I've never tried, but I found img2table https://iamholumeedey007.medium.com/extract-tables-from-images-in-python-ae26a76ba29c

Also, I heard that Azure Document Intelligence is good solution https://azure.microsoft.com/en-us/products/ai-services/ai-document-intelligence
